h2o-py/h2o/explanation/_explain.py

```diff
@@ -1616,64 +1677,90 @@ def model_correlation_heatmap(
+            t.set_color("red")
+    for t in plt.gca().yaxis.get_ticklabels():
+        if _interpretable(t.get_text()):
+            t.set_color("red")
+    fig = plt.gcf()
+    return fig
+
+
+def _check_deprecated_top_n_argument(models, top_n):
+    if top_n is not None:
+        import warnings
+        from h2o.exceptions import H2ODeprecationWarning
+        warnings.warn("Setting the `top_n` parameter is deprecated, use a leaderboard (sub)frame "
+                      "instead, e.g., aml.leaderboard.head({}).".format(top_n), category=H2ODeprecationWarning)
+        models = models.leaderboard.head(top_n)
     else:
-        top_n = len(models)
+        models = models.leaderboard.head(20)
+    return models
 
-    is_classification = frame[models[0].actual_params["response_column"]].isfactor()[0]
 
-    models = models[:min(len(models), top_n)]
-    predictions = np.empty((len(models), frame.nrow),
-                           dtype=np.object if is_classification else np.float)
+def model_correlation(
+        models,  # type: Union[h2o.automl._base.H2OAutoMLBaseMixin, h2o.H2OFrame, List[h2o.model.ModelBase]]
+        frame,  # type: h2o.H2OFrame
+        cluster_models=True,  # type: bool
+        use_pandas=True  # type: bool
+):
+    # type: (...) -> Union[pandas.DataFrame, Tuple[numpy.ndarray, List[str]]]
+    """
+    Get data that are used to build model_correlation_heatmap plot.
+
+    :param models: a list of H2O models, an H2O AutoML instance, or an H2OFrame with a 'model_id' column (e.g. H2OAutoML leaderboard)
+    :param frame: H2OFrame
+    :param cluster_models: if True, cluster the models
+    :param use_pandas: if True, try to return pandas DataFrame. Otherwise return a tuple (correlation_matrix, model_ids)
+    :returns: either pandas DataFrame (if use_pandas == True) or a tuple (correlation_matrix, model_ids)
+    """
+    if _is_automl_or_leaderboard(models):
+        models = list(_get_models_from_automl_or_leaderboard(models))
+    is_classification = frame[models[0].actual_params["response_column"]].isfactor()[0]
+    predictions = []
     with no_progress():
         for idx, model in enumerate(models):
-            predictions[idx, :] = np.array(model.predict(frame)["predict"]
-                                           .as_data_frame(use_pandas=False, header=False)) \
-                .reshape(frame.nrow)
+            predictions.append(model.predict(frame)["predict"])
 
     if is_classification:
         corr = np.zeros((len(models), len(models)))
         for i in range(len(models)):
             for j in range(len(models)):
                 if i <= j:
-                    corr[i, j] = (predictions[i, :] == predictions[j, :]).mean()
+                    corr[i, j] = (predictions[i] == predictions[j]).mean()[0]
                     corr[j, i] = corr[i, j]
     else:
-        corr = np.corrcoef(predictions)
-
+        corr = np.genfromtxt(StringIO(predictions[0].cbind(predictions[1:]).cor().get_frame_data()),
```

Comment:
is `predictions != predictions[0].cbind(predictions[1:])` ?

Reply:
Predictions is `List[H2OFrame]`. `predictions[0].cbind(predictions[1:])` is `H2OFrame`.

Reply:
ok! I made the assumption that `predictions` was already a frame.